src/commands/Tools/emotes.ts

```diff
@@ -0,0 +1,83 @@
+import { MessageEmbed } from 'discord.js';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { UserRichDisplay } from '../../lib/structures/UserRichDisplay';
+import { BrandingColors } from '../../lib/util/constants';
+import { getColor } from '../../lib/util/util';
+
+export default class extends SkyraCommand {
+
+	private kFilter = this.removeNullAndUndefined.bind(this);
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 10,
+			description: language => language.tget('COMMAND_EMOTES_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_EMOTES_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
+			runIn: ['text']
+		});
+	}
+
+	public async run(message: KlasaMessage) {
+		const response = await message.sendEmbed(new MessageEmbed()
+			.setDescription(message.language.tget('SYSTEM_LOADING'))
+			.setColor(BrandingColors.Secondary));
+
+		await this.populateCache(message);
+
+		const animEmotes = message.guild!.emojis.map(emote => emote.animated ? `<a:${emote.name}:${emote.id}>` : undefined).filter(this.kFilter);
+		const staticEmotes = message.guild!.emojis.map(emote => emote.animated ? undefined : `<:${emote.name}:${emote.id}>`).filter(this.kFilter);
```

Comment:
This does 4 loops/iterations (2 .maps and and 2 .filters), it could be done in a single `for of` loop (also agreed to by kyraa)

Reply:
@gc updated